### PR TITLE
Feature/make widget template view reusable

### PR DESCRIPTION
**Pull request by michallaskowski**
This PR will make WidgetTemplateView reusable, that means it can be configured with a view model or layout settings multiple times.

- [x] [SwiftLint](https://github.com/Realm/SwiftLint) is installed on your system
- [x] The project does not generate any warnings or errors
- [x] The tests are passing
- [x] Branch name is in line with [GitFlow](http://jeffkreeftmeijer.com/2010/why-arent-you-using-git-flow/)

### CODE REVIEW

**Comment by ochococo** [comment]
I would like to ask you to consider this:
https://github.com/macoscope/GrandCentralBoard/issues/164

While you at this, just move those files from Core to Utilities. WDYT?

**Comment by michallaskowski** [comment]
@ochococo I wanted to do it in another PR (or I think @krris wanted to do it too), cause it also touches widgets, and wanted to keep it small. I had this PR almost finished before this issue was created.

**Comment by ochococo** [comment]
@michallaskowski Got it, makes sense.

File: GCBCore/Views/WidgetTemplateView.swift [hunk 1]
```diff
@@ -85,47 +85,52 @@ public class WidgetTemplateView: UIView {
...
             containerTopToHeaderBottomConstraint.active = true
             containerTopConstraint.active = false
         }
+        currentLayoutSettings = layoutSettings
 
         guard let contentView = contentView else { return }
-        contentView.frame = UIEdgeInsetsInsetRect(contentView.bounds, layoutSettings.contentMargin)
+        contentView.frame = UIEdgeInsetsInsetRect(contentViewContainer.bounds, layoutSettings.contentMargin)
+        contentView.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
+        setNeedsLayout()
```
Comment (krris) [inline GCBCore/Views/WidgetTemplateView.swift]: Do we need this? As I understand changing `frame` should also trigger layout update during next update cycle. Correct me if I'm wrong.

**Comment by krris** [comment]
Works nice 🎾  @michallaskowski

File: GCBCore/Views/WidgetTemplateView.swift (hunk 1 above)
Comment (michallaskowski) [inline GCBCore/Views/WidgetTemplateView.swift]: It's true, if `autoresizeSubviews` is true. And here it is, unless a class using this view set it to false. But if it did, then for some reason it should not layout. So yeah, we can remove it ;)

**Pull request merged**